geniushubclient/zone.py

```diff
@@ -221,18 +344,257 @@ def footprint_schedule(raw_json) -> Dict:
 
         return self._data
 
+    def update_state(self, json):
+        """Parse the json and use it to populate the state data class."""
+        try:
+            self._state.is_active = json["bIsActive"]
+            self._state.is_requesting_heat = json["bOutRequestHeat"]
+
+            iType = json["iType"]
+            if iType in [ZONE_TYPE.ControlSP, ZONE_TYPE.TPI]:
+                # some zones have a fPV without raw_json["activeTemperatureDevices"]
+                self._state.temperature = json["fPV"]
+                self._state.set_point = json["fSP"]
+
+            if iType == ZONE_TYPE.Manager:
+                if json["fPV"]:
+                    self._state.temperature = json["fPV"]
+
+            mode = json["iMode"]
+            self._state.mode = mode
+            self._state.mode_name = IMODE_TO_MODE[mode]
+
+        except (AttributeError, LookupError, TypeError, ValueError):
+            _LOGGER.exception("Failed to set Zone %s state.", self.id)
+
+    def update_properties(self, json):
+        """Parse the json and use it to populate the properties data class."""
+        try:
```

Comment:
It's a fair comment on only having these on the complex functions. I saw that those original code blocks had them in, so added them to all the update methods to be consistent. Also, as this is all new, I thought that it was better for me to err on the side of caution with all these.

I'm happy to reduce the number, if you think that's better. Would you want one over-arching one in the update method, as well as the two complex functions?

Reply:
I don't profess to be a Python expert (and I know nothing about java, et al.), but I wonder if there is a difference in philosophy about the these try/catch, try/except clauses? Anyway...

Let's be brave - wrap complex code (timer, footprint), but no over-arching try/except. We can do extensive testing as is essentially Summer now - anyone who is risk-averse can stick to the PyPi library.

Reply:
No, in C# you really shouldn't use too many try/catch blocks, as exceptions are intensive to create (the whole stack needs to be unwound to create the stack trace, for starters). As I said, I only added them so that all the update methods were consistent, but they were a pain in the arse for the unit tests, as they would catch the errors, so tests could pass when they shouldn't.

I'll just keep the try/catch code in the schedule update methods.

Reply:
Updated to remove the extra error catching.